src/workspace.ts

```diff
@@ -0,0 +1,81 @@
+import { TreeDataProvider, EventEmitter, TreeItemCollapsibleState, TreeItem, ProviderResult, Event } from 'vscode';
+import { globalenv } from './session';
+
+export interface WorkspaceAttr {
+    [key: string]: {
+        class: string[];
+        type: string;
+        length: number;
+        str: string;
+    }
+}
+
+export class WorkspaceDataProvider implements TreeDataProvider<WorkspaceItem> {
+	private _onDidChangeTreeData: EventEmitter<WorkspaceItem | undefined | null | void>
+		= new EventEmitter<WorkspaceItem | undefined | null | void>();
+	readonly onDidChangeTreeData: Event<WorkspaceItem | undefined | null | void>
+		= this._onDidChangeTreeData.event;
+
+	refresh(): void {
+		this.data = globalenv;
+		this._onDidChangeTreeData.fire();
+	}
+
+	data: WorkspaceAttr;
+
+	constructor() {}
+
+	getTreeItem(element: WorkspaceItem): TreeItem | Thenable<TreeItem> {
+		return element;
+	}
+
+	getChildren(element?: WorkspaceItem): ProviderResult<WorkspaceItem[]> {
+		return Promise.resolve(
+			this.getWorkspaceItems(this.data));
+	}
+
+	private getWorkspaceItems(data: any): WorkspaceItem[] {
+		const toItem = (
+			key: string,
+			rClass: string[],
+			type: string,
+			typeDetailed: string,
+			length: number
+		): WorkspaceItem => {
+			return new WorkspaceItem(
+				key,
+				rClass,
+				type,
+				typeDetailed,
+				length,
+				TreeItemCollapsibleState.None
+			);
+		};
+
+		const items = data ? Object.keys(data).map((key) =>
+			toItem(
+				key,
+				data[key].class,
+				data[key].type,
+				data[key].str,
+				data[key].length
+			)) : [];
+
+		return items;
+	}
```

Comment:
We should probably consider sorting the elements by name and/or by type here

Reply:
I've sorted the elements by name and type like so (2a97b29):

- key: string,
- rClass: string,
- str: string,
- type: string,
- length: number

Is this what you had in mind?

Reply:
Sorry that comment was worded stupidly. I meant the items shown in the workspace viewer. Currently, assigning e.g. 
``` r
> a <- 1
> c <- 3
> b <- 2
> l <- list(1,2,3)
> ll <- list(3,4,5)
> h <- 99
> x <- 4
```
will be shown in the workspace view as
```
x num 4
ll List of 3
a num 1
b num 2
c num 3
h num 99
l List of 3
```

I was trying to suggest either sorting by variable name:
```
a num 1
b num 2
c num 3
h num 99
l List of 3
ll List of 3
x num 4
```
and/or variable type:
```
a num 1
b num 2
c num 3
h num 99
x num 4

l List of 3
ll List of 3
```

Sorry again for being so unclear

Reply:
In RStudio, the environment pane looks like

<img width="864" alt="image" src="https://user-images.githubusercontent.com/4662568/100740953-77460580-3414-11eb-9c48-c7705a0ecb83.png">

Not sure if we could do something similar?

Reply:
@ManuelHentschel Ah, no, not at all, that makes a lot more sense in retrospect (although sorting the inputs is probably helpful for reading the code anyway). 

Sorting should be easy enough, with something like `return items.sort((a, b) => a.label.localeCompare(b.label) || a.contextValue.localeCompare(b.contextValue));` where contextValue is defined as the type in the WorkspaceItem constructor.

@renkun-ken The tree view is a bit limited in terms of customisation/UI, but I think it should be possible if I create parent elements "Data" and "Values" and then make the make the tree items children of them. I might have to fiddle around with this a bit!

Reply:
>I create parent elements "Data" and "Values" and then make the make the tree items children of them. I might have to fiddle around with this a bit!

I think there's no need to do make it complicated. Just find `data.frame`, `list`, and `environment` as data variables and put them on top, and data variables and non-data variables are sorted alphabetically, respectively.

Reply:
![image](https://user-images.githubusercontent.com/60372411/100946666-b7cc7e80-34fb-11eb-91dc-9deeb701971b.png)

Sorting should now

- Prioritise data at the top
- Then sort by name alphabetically
- Then sort by class alphabetically

Reply:
If sort by name, then there won't be duplicates, why "Then sort by class alphabetically"?

Reply:
Err... good point. Oops. I'll fix that up